service/src/test/java/bio/terra/catalog/service/DatasetServiceTest.java

```diff
@@ -129,6 +130,16 @@ void listDatasets() {
     assertThat(tdrJson.get("accessLevel").asText(), is(String.valueOf(DatasetAccessLevel.OWNER)));
   }
 
+  @Test
+  void getMetadata() throws JSONException {
+    when(datarepoService.getRole(SOURCE_ID)).thenReturn(DatasetAccessLevel.OWNER);
+    when(datarepoService.getDataset(SOURCE_ID))
+        .thenReturn(new StorageSystemInformation().datasetAccessLevel(DatasetAccessLevel.OWNER));
+    when(datasetDao.retrieve(any())).thenReturn(tdrDataset);
```

Comment:
Shouldn't this be
```suggestion
    when(datasetDao.retrieve(tdrDataset.id())).thenReturn(tdrDataset);
```

Also, is there a reason to use the TDR service here? Why not use the external service, which would let you use `mockDataset()`?

I think overall this test code should be overhauled to use the same service for all tests (which can be any service at this point, since at this layer no service has any special behavior) except for a test that's specifically testing the mapping of storage system type to service, or of handling multiple services at the same time (such as in `listDatasets()`).